### DISCUSSION TITLE
Magic Generator Property Names

### DISCUSSION THREAD

**Discussion by ImCesar** [nrwl/nx-console#1091] [Q&A] [answered]
There seems to be some property keys you can use in your generator schema that have special interactions with the NX Console. Is there anyway to get a list of these or have them documented somewhere?

One example is if you name a property `project` then it will give you a list of all projects in your workspace.json. It even allows you to type and filter that list. Another one is path, if you right click in the file explorer and run the generator that way it will auto populate the path (along with the project). 

These might be the only ones, but I was curious if there were more and where we could find them.

**Comment by Cammisuli** [comment, marked as answer]
Those two that you mention are pretty much it.

The project property is used with the core nx generators and even the angular schematics. I tried to look this up in some documentation, but it's not mentioned anywhere, so I apologize for that. We'll try and docs updates to include the project property.

The path/directory property is just something we handle ourselves because it's just a common pattern.
